src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,93 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
+                var finalRoot = LocalFunction
+                    ? InsertLocalFunction()
+                    : InsertNormalMethod();
 
+                return await documentWithUpdatedCallSite.WithSyntaxRootAsync(finalRoot, cancellationToken).ConfigureAwait(false);
+
+                SyntaxNode InsertLocalFunction()
+                {
+                    // Now, insert the local function.
                     var info = codeGenerationService.GetInfo(
-                        new CodeGenerationContext(
-                            generateDefaultAccessibility: false,
-                            generateMethodBodies: true),
+                        new CodeGenerationContext(generateDefaultAccessibility: false),
                         Options,
-                        callSiteDocument.Project.ParseOptions);
+                        document.Project.ParseOptions);
+
+                    var localMethod = codeGenerationService.CreateMethodDeclaration(newMethodDefinition, CodeGenerationDestination.Unspecified, info, cancellationToken);
+
+                    // Find the destination for the local function after the callsite has been fixed up.
+                    var destination = InsertionPoint.With(documentWithUpdatedCallSite).GetContext();
+                    var updatedDestination = codeGenerationService.AddStatements(destination, new[] { localMethod }, info, cancellationToken);
 
-                    var localMethod = codeGenerationService.CreateMethodDeclaration(result.Data, CodeGenerationDestination.Unspecified, info, cancellationToken);
-                    newContainer = codeGenerationService.AddStatements(destination, new[] { localMethod }, info, cancellationToken);
+                    var finalRoot = documentWithUpdatedCallSite.Root.ReplaceNode(destination, updatedDestination);
+                    return finalRoot;
                 }
-                else
+
+                SyntaxNode InsertNormalMethod()
                 {
-                    var previousMemberNode = GetPreviousMember(callSiteDocument);
+                    var syntaxKinds = document.GetLanguageService<ISyntaxKindsService>();
+
+                    // Find the destination for the new method after the callsite has been fixed up.
+                    var mappedMember = this.InsertionPoint.With(documentWithUpdatedCallSite).GetContext();
+                    mappedMember = mappedMember.Parent?.RawKind == syntaxKinds.GlobalStatement
+                        ? mappedMember.Parent
+                        : mappedMember;
 
                     // it is possible in a script file case where there is no previous member. in that case, insert new text into top level script
-                    destination = previousMemberNode.Parent ?? previousMemberNode;
+                    var destination = mappedMember.Parent ?? mappedMember;
```

Comment:
the logic here is very wonky/confusing, but is maintained from before.  i really don't know this code needs to muck with global-statements/scripts but i'm keeping it as is.  personally, i think it's likely that the get-insertion-node logic for methods is the part taht actually needs to be revised to properly get the right insertion point so that we don't need to compensate for the issue over here.